source/module_io/read_rho.cpp

```diff
@@ -1,7 +1,29 @@
 #include "module_io/rho_io.h"
-#include "module_hamilt_pw/hamilt_pwdft/global.h"
+#include "module_base/global_variable.h"
 
-bool ModuleIO::read_rho(const int &is, const std::string &fn, double* rho, int &prenspin) //add by dwan
+#ifdef __MPI
+bool ModuleIO::read_rho(Parallel_Grid &Pgrid, const int &is,
```

Comment:
I suggest you use "Parallel_Grid * Pgrid" but not "Parallel_Grid & Pgrid".
Then you can separate MPI case and no-MPI case by this code:
```
Parallel_Grid * pgrid_tmp = nullptr;
#ifdef __MPI
pgrid_tmp = &GlobalC::Pgrid;
#endif
```

Reply:
You can also use :
```
bool ModuleIO::read_rho(
#ifdef __MPI
                    Parallel_Grid &Pgrid,
#endif
                    const int &is,
		    const int &nspin,
		    const std::string &fn,
		    double* rho,
		    int& nx,
		    int& ny,
		    int& nz,
		    double& ef,
		    UnitCell& ucell,
		    int &prenspin)
```